### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by Shevtsooov**
[DEMO LINK](https://Shevtsooov.github.io/react_dynamic-list-of-goods/)

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,27 +1,55 @@
...
+      throw new Error();
+    }
+  };
 
-    <button type="button" data-cy="first-five-button">
-      Load 5 first goods
-    </button>
+  const handleGetAll = useCallback(() => loadGoods(getAll), []);
+  const handleGet5First = useCallback(() => loadGoods(get5First), []);
+  const handleGetRedGoods = useCallback(() => loadGoods(getRedGoods), []);
```
Comment (dffUqp) [inline src/App.tsx:22]: I think we need to remove `useCallback` here, because we prevent render for 0 components

File: src/App.tsx [hunk 2]
```diff
@@ -1,27 +1,55 @@
...
+export const App: React.FC = () => {
+  const [filteredGoods, setFilteredGoods] = useState<Good[]>([]);
 
-    <button type="button" data-cy="all-button">
-      Load all goods
-    </button>
+  const loadGoods = (callback: () => Promise<Good[]>) => {
+    try {
+      callback()
+        .then(goods => setFilteredGoods(goods));
```
Comment (dffUqp) [inline src/App.tsx:14]: let's rewrite this with async/await

**Review by dffUqp — CHANGES_REQUESTED** [review]
LGTM:fire: but needs a few fixes:upside_down_face:

**Comment by Shevtsooov** [comment]
@dffUqp Thanks for the review, the changes have been applied)

**Review by dffUqp — APPROVED** [review]
🚀

**Pull request closed without merging**